backbone/src/main/java/org/researchstack/backbone/model/survey/QuestionSurveyItem.java

```diff
@@ -0,0 +1,188 @@
+package org.researchstack.backbone.model.survey;
+
+import com.google.gson.annotations.SerializedName;
+
+import org.researchstack.backbone.step.QuestionStep;
+import org.researchstack.backbone.step.Step;
+import org.researchstack.backbone.utils.SurveyFactory;
+
+import java.util.List;
+
+/**
+ * Created by TheMDP on 12/31/16.
+ */
+
+public class QuestionSurveyItem<T extends Object> extends SurveyItem<T> {
+
+    @SerializedName("questionStyle")
+    public boolean questionStyle;
+    @SerializedName("placeholderText")
+    public String placeholderText;
+    @SerializedName("optional")
+    public boolean optional;
+    @SerializedName("range")
+    public SurveyRange range;
+
+    @SerializedName("expectedAnswer")
+    public boolean expectedAnswer; // Does this need to be a generic type?
+
+    public boolean isValidQuestionItem() {
+        return identifier != null && type.isQuestionSubtype();
+    }
+
+    public boolean isBooleanToggle() {
+        return type == SurveyItemType.QUESTION_TOGGLE;
+    }
+
+    public boolean isCompoundStep() {
+        return isBooleanToggle() || type == SurveyItemType.QUESTION_COMPOUND;
+    }
+
+    // TODO: moved to SubtaskQuestionSurveyItem
+//    public SubtaskStep createSubtaskStep(SurveyFactory factory) {
+//        if (items == null || items.isEmpty()) {
+//            throw new IllegalStateException("A subtask step requires items, since the items are the steps");
+//        }
+//
+//        List<Step> steps = factory.createSteps(items, true);
+//        SubtaskStep step = usesNavigation() ?
+//            // TODO: Create NavigationSubtaskStep, for now just return subtask
+//            new SubtaskStep(identifier, steps) :
+//            new SubtaskStep(identifier, steps);
+//        return step;
+//    }
+
+    // TODO: moved somewhere else, probably in task creation with navigation rules
+//    boolean usesNavigation() {
+//        if (skipIdentifier != null || rulePredicate != null) {
+//            return true;
+//        }
+//        if (items == null) {
+//            return false;
+//        }
+//        for (SurveyItem item : items) {
+//            if (item instanceof QuestionSurveyItem &&
+//               ((QuestionSurveyItem)item).rulePredicate != null)
+//            {
+//                return true;
+//            }
+//        }
+//        return false;
+//    }
+
+    public Step createQuestionStep(boolean isSubtaskStep, SurveyFactory factory) {
+        // Factory method for determining the proper type of form-style step to return
+        // the ORKQuestionStep and ORKFormStep have a different UI presentation
+        QuestionStep step = null;
+        if (isBooleanToggle()) {
+            // TODO finish coding
+            //step = new QuestionStep()
+        }
+        return step;
+    }
+
+//
+//    func createFormStep(isSubtaskStep: Bool, factory: SBASurveyFactory? = nil) -> ORKStep {
```

Comment:
All this will probably be moved into the SurveyFactory, but for now I'm leaving it for reference in where it was in the iOS library.